phone_verify/backends/base.py

```diff
@@ -49,7 +49,21 @@ def generate_session_token(cls, phone_number):
         identifying a particular device in subsequent calls.
         """
         data = {"phone_number": phone_number, "nonce": random.random()}
-        return jwt.encode(data, django_settings.SECRET_KEY).decode()
+        encoded_str = jwt.encode(data, django_settings.SECRET_KEY)
+        if type(encoded_str) is bytes:
+            # jwt v2.0.0 or greater: `jwt.encode()` no longer return <class 'bytes'> 
+            # instead it return <class 'str'> and that's why `jwt.encode().decode()` fails
+            # To support jwt v1.7.1 or greater: if `jwt.encode()` return <class 'bytes'>
+            # convert it to <class 'str'>
+            # Details: https://github.com/jpadilla/pyjwt
+            try:
+                encoded_str = str(encoded_str, encoding='utf-8')
+            except TypeError:
+                try:
+                    encoded_str = encoded_str.decode()
+                except AttributeError:
+                    pass
```

Comment:
Why are we catching `AttributeError` here? In what case does it cause an `AttributeError`?

Reply:
> Why are we catching `AttributeError` here? In what case does it cause an `AttributeError`?

Thanks for the review.
It was a mistake, at first the `if` condition was different in my local (before commit), but later I updated the `if` conditional block.
```python
encoded_str = jwt.encode(data, django_settings.SECRET_KEY)
if type(encoded_str) is not str:
    # jwt v2.0.0 or greater: `jwt.encode()` no longer return <class 'bytes'> 
    # instead it return <class 'str'> and that's why `jwt.encode().decode()` fails
    # To support jwt v1.7.1 or greater: if `jwt.encode()` return <class 'bytes'>
    # convert it to <class 'str'>
    # Details: https://github.com/jpadilla/pyjwt
    try:
        encoded_str = str(encoded_str, encoding='utf-8')
    except TypeError:
        try:
            encoded_str = encoded_str.decode()
        except AttributeError:
            pass
return encoded_str
```
In that case `PyJWT v2.0.1` can through `AttributeError`  because str object does not have decode().  But I noticed It will not happen.  But I have forgotten to remove that line.

It should be removed.

By the way, it was my first pull request. I am not good at testing. But it worked for me. Please apologize for my mistake.
I am updating this.

Reply:
Thank you for coming forward to contribute to this library. I really appreciate it. I've mentioned some comments and suggestions inline. Please have a look at them.